### DISCUSSION TITLE
fieldArray append function requires valid defaultValues

### DISCUSSION THREAD

**Discussion by boschr** [react-hook-form/react-hook-form#9639] [Q&A]
Since version `7.34` where the `Partial` wrapper is removed in https://github.com/react-hook-form/react-hook-form/pull/8660/files#diff-ee762f75b15dde4fde77049f36e387fa0421f37dc1c29dad72fceba8a23c0e4d is it now required to put valid defaultValues to the `append` function. This does not work when f.e. appending a new row with a default empty select which value is required in the schema.

Example:
```
const schema = z.object({
  array: z.array(
    z.object({
      field: z.enum(["Value1", "Value2"]),
    })
  ),
});

const emptyInvalidForm = { field: null };
const emptyValidForm = { field: "Value1" };

const Form: FC = () => {
  const { control } = useFormContext();
  const { fields, append } = useFieldArray({
    control,
    name: "array",
  });

  const handleClick = useCallback(() => {
    // Valid
    append(emptyValidForm);

    // Invalid
    append(emptyInvalidForm);
  }, []);

  return (
    <>
      {fields.map((field, index) => (
        <Select key={field.id} name={`array.${index}.field`} />
      ))}
      <Button onClick={handleClick}>Add</Button>
    </>
  );
};
```

**Comment by leapful** [comment]
@boschr 

I think you are misunderstanding zod documentation.

By default, field definition of zod is not optional, you need to put `.optional()` to make it not mandatory. Furthermore, when preparing enum value, it should be the enum from zod infer, not a plain string.

Working demo: https://codesandbox.io/s/adoring-rain-nivovd?file=/src/App.tsx

**Reply by boschr** [reply]
@leapful thanks you for your reply. I do understand the ZOD documentation and i know that everything is mandatory by default. Probably my question is not 100% clear. The field should be mandatory in the form, but i don't want the "select" input to be prefilled with an enum value when I add a new "item" to the fieldArray. I want to add an empty item.

Also the enum in the example is a bad example. In the real situation we're using it in this way (the enum is generated from api schema's):
```
z.object({
  field: z.nativeEnum(FieldType),
})```
